src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5229,6 +5301,9 @@ bool StorageReplicatedMergeTree::waitForShrinkingQueueSize(size_t queue_size, UI
 
     /// Let's fetch new log entries firstly
     queue.pullLogsToQueue(getZooKeeper());
+    /// This is significant, because the execution of this task could be delayed at BackgroundPool.
+    /// And we force it to be executed.
+    queue_task_handle->wake();
```

Comment:
But why? `waitForShrinkingQueueSize` called in `SYSTEM SYNC REPLICA` query only. You may notice, that `pullLogsToQueue` itself triggers this handle https://github.com/clickhouse/ClickHouse/blob/master/src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp#L563-L564.

Reply:
But it triggers this handle only if `log_entries` set is not empty https://github.com/ClickHouse/ClickHouse/blob/8d9240d3e9f34648c0cdbfe06a86323f378c1e63/src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp#L461
Is it possible that `log_entries` set is empty, but we have some in local queue? For instance some task is stuck because of errors?